File: all-submissions/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] chArray = s.toCharArray();
        int left =0;
        int right =0;
        Map<Character, Integer> freqMap = new HashMap<>();
        int longest = 0;
        while(right < chArray.length){
            freqMap.put(chArray[right], freqMap.getOrDefault(chArray[right], 0) + 1);
            if(freqMap.get(chArray[right]) > 1){
                while(freqMap.get(chArray[right]) > 1){
                    freqMap.put(chArray[left], freqMap.get(chArray[left])-1);
                    left++;
                }
            }
            longest = Math.max(longest, right-left+1);
            right++;
        }
        return longest;
    }
}

File: all-submissions/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_dict = {}
        left = 0
        right = 0
        max_length = 0
        while(right < len(s)):
            char_dict[s[right]] = char_dict.get(s[right], 0) + 1
            while left < right and char_dict[s[right]] > 1:
                char_dict[s[left]] -=1 
                left +=1
                if char_dict[s[left]] == 0:
                    del char_dict[s[left]]
            max_length = max(max_length, right - left +1)
            right += 1
        return max_length
